src/main/scala/net/ruippeixotog/scalascraper/browser/HtmlUnitBrowser.scala

```diff
@@ -123,6 +123,11 @@ object HtmlUnitBrowser {
     def tagName = underlying.getTagName
     def parent = Option(underlying.getParentNode).collect { case elem: DomElement => HtmlUnitElement(elem) }
     def children = underlying.getChildElements.map(HtmlUnitElement)
+    def siblings = {
+      val previousSiblings = Stream.iterate(underlying.getPreviousElementSibling)(_.getPreviousElementSibling).takeWhile(_ != null)
```

Comment:
I don't believe there is much difference between the two approaches. Both will call `getPreviousElementSibling` the same amount of times and, as `Stream` is lazy, the resulting collection will always have the filtered elements only. At most, you'd be avoiding an extra `Stream.Cons` node, which doesn't warrant the slightly worse code style.